docs/docs/guides/tracking/costs.md

```diff
@@ -0,0 +1,163 @@
+# Costs
+
+Costs are an evolving feature of Weave. Currently, the Weave UI calculates costs based on the number of tokens used and the model used, and displays them in the UI, using a static cost file.
+
+In the future, Weave will use a custom cost model, which can be set up by the user. Currently, custom costs can be set up by the user, and fetched by the user in the client, but costs in the UI are still based on token usage, model, and static cost file.
+
+## Adding a custom cost
+
+You can add a custom cost by using the [`add_costs`](/reference/python-sdk/weave/trace/weave.trace.weave_client#method-add_costs) method.
+The two required fields are `prompt_token_cost` and `completion_token_cost`.
+You can also set `effective_date` to a datetime, to make the cost effective at a specific date, this defaults to the current date.
+
+```python
+import weave
+from datetime import datetime
+
+client = weave.init("my_custom_cost_model")
+
+client.add_costs({
```

Comment:
This method is `WeaveClient::add_costs(self, costs: Dict[str, CostCreateInput]) -> CostCreateRes`

Using CostCreateInput here is not great - i think it would be better to have each of the fields of CostCreateInput map to a parameter. As implemented, the user would need to import CostCreateInput to construct the query. Your example in docs uses a plain dictionary, but if type checked it would be wrong.